pypesto/hierarchical/optimal_scaling_approach/__init__.py

```diff
@@ -0,0 +1,10 @@
+"""
```

Comment:
would a more descriptive (and shorter) name of the package be `ordinal` instead of `optimal_scaling_approach`?

Reply:
I was thinking about this for a while and later discussed it with Daniel. 
On one hand, the method or approach that's implemented is the optimal scaling one and it just deals with ordinal data. So it would be more correct to name it like it is in this sense.

On the other hand, it would make it much simpler if it was all just called ordinal, since it's the specific application to ordinal data, and the optimal scaling currently cannot be used for some other data.

But then when I'll implement the optimal scaling approach for censored data. I'll probably just extend the optimal scaling approach to detect the censored data and include it (in the same module).
Or I can separate it and call that one censored (might be easier to read and understand what's handling what).

So not sure what's best. Open to suggestions/other namings.

Reply:
So none of the names is good :sweat_smile: My recommendations:

1. Keep it short. Thus, definitely remove the `_approach`.
2. Whether to call it `ordinal` or `optimal_scaling` (or `rescaling` or `scaling` or `scale`) I would let depend on whether you intend to implement the censored case in the same submodule (if it uses much of the same base code) (--> optimal_scaling) or implement it as a separate submodule (-->ordinal).

Reply:
Yeah, true,`_approach` is definitely not important.

Ok, I'll leave further renaming for when I decide how's best to implement censored data.

Reply:
Further, on the point of naming: I saw a pattern in pypesto that even though a python file is specific to a module it still has a general name, for instance:
petab checks for hierarchical are in the file hierarchical/petab.py and not  hierarchical/hierarchical_petab.py
or the parameter is hierarchical/parameter.py and not something like hierarchical/inner_parameter.py

So should all things in the optimal_scaling folder be just calculator, solver, parameter and problem, or be left as optimal_scaling_calculator, ... etc.

Reply:
As the fully qualified path would be `pypesto.hierarchical.{optimal_scaling etc}.solver`, the additional `optimal_scaling` can be omitted indeed. Note: Actually, as all important exports are given in `pypesto.hierarchical.optimal_scaling.__init__`, nobody should ever directly refer to the file name, but only use what is in `__init__`. This could be enforced more strictly by setting files to private, `pypesto.hierarchical.optimal_scaling._solver`, but this we have done nowhere in pypesto either, so one could do it here but also fine without.

Reply:
Ok, will rename to only `calculator`, `solver`, `parameter` and `problem`, looks nicer. 
I won't add _ as it wasn't added to anything else, for consistency of the whole package.